optuna/storages/_redis.py

```diff
@@ -607,3 +643,48 @@ def _check_trial_id(self, trial_id: int) -> None:
 
         if not self._redis.exists(self._key_trial(trial_id)):
             raise KeyError("study_id {} does not exist.".format(trial_id))
+
+    def record_heartbeat(self, trial_id: int) -> None:
+        study_id = self.get_study_id_from_trial_id(trial_id)
+        self._redis.hset(
+            self._key_study_heartbeats(study_id), str(trial_id), pickle.dumps(time.time())
```

Comment:
Can you use the current time of DB side, not client side? c.f. https://github.com/optuna/optuna/blob/master/optuna/storages/_rdb/storage.py#L1197